### PR TITLE
[Feature] Support materialized view for external table (step 1/3)

### PR DESCRIPTION
**Pull request by Youngwb**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

#11116 
#9792

## Problem Summary(Required) ：

In this PR, There are two main feature:
1. support create mv based on external (hive/hudi) **unpartitioned** table 
2. support create mv select data from different database

Major changes:
1. Use BaseTableInfo instead of baseTableIds to represent MV-dependent base tables
2. Where OlapTable is heavily dependent, use Table instead
3. Because dbId and tableId are not available for external table, use TableIdentifier to uniquely identify a table

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by Youngwb** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 121 / 176 (68.75%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|28|61|45.90%|[81, 82, 83, 84, 85, 86, 89, 90, 92, 97, 98, 100, 105, 109, 113, 117, 132, 137, 138, 139, 141, 149, 377, 445, 446, 448, 449, 450, 452, 453, 460, 464, 465]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|1|2|50.00%|[171]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[176]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|12|17|70.59%|[131, 132, 133, 145, 146]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|65|79|82.28%|[320, 336, 401, 460, 465, 474, 531, 532, 533, 538, 539, 540, 548, 552]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|7|8|87.50%|[174]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStatement.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|5|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 121 / 177 (68.36%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/catalog/IcebergTable.java|0|1|00.00%|[107]|
|:large_blue_circle:|com/starrocks/catalog/MaterializedView.java|28|61|45.90%|[81, 82, 83, 84, 85, 86, 89, 90, 92, 97, 98, 100, 105, 109, 113, 117, 132, 137, 138, 139, 141, 149, 377, 445, 446, 448, 449, 450, 452, 453, 460, 464, 465]|
|:large_blue_circle:|com/starrocks/catalog/HudiTable.java|1|2|50.00%|[171]|
|:large_blue_circle:|com/starrocks/catalog/HiveTable.java|1|2|50.00%|[176]|
|:large_blue_circle:|com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java|12|17|70.59%|[131, 132, 133, 145, 146]|
|:large_blue_circle:|com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java|65|79|82.28%|[320, 336, 401, 460, 465, 474, 531, 532, 533, 538, 539, 540, 548, 552]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|7|8|87.50%|[174]|
|:large_blue_circle:|com/starrocks/sql/ast/CreateMaterializedViewStatement.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|5|5|100.00%|[]|

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java [hunk 1]
```diff
@@ -114,35 +117,39 @@ public Void visitCreateMaterializedViewStatement(CreateMaterializedViewStatement
...
-                            "Materialized view do not support table: " + table.getName() +
-                                    " do not exist in database: " + db.getOriginName());
-                }
-                if (!(table instanceof OlapTable)) {
-                    throw new SemanticException(
-                            "Materialized view only supports olap table, but the type of table: " +
-                                    table.getName() + " is: " + table.getType().name());
+            tableNameTableMap.forEach((tableNameInfo, table) -> {
+                if (table == null) {
+                    throw new SemanticException("Materialized view do not support table: " + tableNameInfo.getTbl() +
```
Comment (mofeiatwork) [inline fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:131]: How about simplify the message as 'referenced table not exist'

**Review by murphyatwork — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java (hunk 1 above)
Comment (Youngwb) [reply fe/fe-core/src/main/java/com/starrocks/sql/analyzer/MaterializedViewAnalyzer.java:131]: ok， I will simplify the message in next PR

**Pull request merged**